syntax = "proto3";

package tt.scaleout.validation;

message TrafficParams {
    uint32 packet_size_bytes = 1;
    uint32 data_size = 2;
}

message EthernetMetrics {
    uint32 retrain_count = 1;
    uint32 crc_error_count = 2;
    uint64 corrected_codeword_count = 3;
    uint64 uncorrected_codeword_count = 4;
}

message LinkStatus {
    EthernetMetrics ethernet_metrics = 1;
    TrafficParams traffic_params = 2;
    uint32 num_mismatched_words = 3;
}

message EthChannelIdentifier {
    string host = 1;
    uint64 asic_id = 2;
    uint64 tray_id = 3;
    uint64 asic_location = 4;
    uint32 channel = 5;
}

message EthernetLinkMetrics {
    EthChannelIdentifier channel_identifier = 1;
    LinkStatus link_status = 2;
}

message EthernetLinkMetricsList {
    repeated EthernetLinkMetrics link_metrics = 1;
}

message EthChannelIdentifierList {
    repeated EthChannelIdentifier exit_nodes = 1;
}

message ResetPair {
    uint32 src_rank = 1;
    uint32 dst_rank = 2;
}

message ResetPairList {
    repeated ResetPair reset_pairs = 1;
}
